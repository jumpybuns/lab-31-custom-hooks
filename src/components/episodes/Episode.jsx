import React from 'react';
import PropTypes from 'prop-types';

const Episode = ({ name, url }) => (
  <figure>
    <h1>{name}</h1>
    <h2>{url}</h2>
  </figure>
);

Episode.propTypes = {
  name: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default Episode;
