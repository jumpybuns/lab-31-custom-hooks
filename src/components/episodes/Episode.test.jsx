import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Episode from './Episode';

describe('Episode component', () => {
  afterEach(() => cleanup());
  it('renders Episode', () => {
    const { asFragment } = render(<Episode name="Arnold" url="http" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
