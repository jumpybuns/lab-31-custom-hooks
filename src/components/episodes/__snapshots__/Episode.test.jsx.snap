// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Episode component renders Episode 1`] = `
<DocumentFragment>
  <figure>
    <h1>
      Arnold
    </h1>
    <h2>
      http
    </h2>
  </figure>
</DocumentFragment>
`;
